### PR TITLE
fix(transform): preserve strikethrough around auto-linked URLs #DOCSTOOLS-5243

### PR DESCRIPTION
**Pull request by kadymov**
## Description

- The handling of auto‑links inside strikethrough text has been fixed: `~~https://...~~` now displays correctly as a strikethrough link.
- An inline rule has been added that prevents `linkify` from consuming the closing `~~`.
- Support for URLs with permissible `~` characters at the end has been retained.
- Regression tests for basic and edge cases have been added.
- The fix has been verified by tests, typecheck, build, and visually via testpack.

Related issue: #DOCSTOOLS-5243

### CODE REVIEW

File: src/transform/plugins/linkify-strikethrough.ts [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    let end = value.length;
+
+    while (end > 0 && value[end - 1] === '*') {
+        end--;
+    }
+
+    return value.slice(0, end);
+}
+
+export default function linkifyStrikethrough(md: MarkdownIt) {
```
Comment (separatrixxx) [inline src/transform/plugins/linkify-strikethrough.ts]: А всё-таки, какое тут преимущество от дефолтного импорта? Мой перфекционист просто хочет, чтобы всё было одинаково)

**Review by goldserg — APPROVED** [review]
